frontend/src/app/history-query/edit-history-query/edit-history-query.component.html

```diff
@@ -0,0 +1,185 @@
+<div class="oi-padded-container">
+  <div class="d-flex justify-content-between">
+    <div class="oi-header-container">
+      <h1 *ngIf="historyQuery" translate="history-query.edit-title" [translateParams]="{ name: historyQuery.name }"></h1>
+      <h1 *ngIf="!historyQuery" translate="history-query.create-title"></h1>
+    </div>
+  </div>
+
+  <div class="row justify-content-center">
+    <div class="col-md-10 col-lg-8">
+      <form [formGroup]="historyQueryForm" (ngSubmit)="save()" id="history-query-form" class="mt-3">
+        <!-- General settings -->
+        <h2 translate="history-query.general-settings"></h2>
+        <div class="row mb-3">
+          <!-- Name -->
+          <div class="col-4">
+            <div class="form-group">
+              <label for="history-query-name" translate="history-query.name"></label>
+              <input formControlName="name" id="history-query-name" class="form-control" />
+              <val-errors controlName="name"> </val-errors>
+            </div>
+          </div>
+
+          <!-- Enabled -->
+          <div class="col-4">
+            <div class="form-group">
+              <label for="history-query-enabled" translate="history-query.enabled"></label>
+              <div class="form-check form-switch">
+                <input formControlName="enabled" id="history-query-enabled" type="checkbox" class="form-check-input" />
+              </div>
+            </div>
+          </div>
+        </div>
+        <div class="row mb-3">
+          <!-- Description -->
+          <div class="form-group">
+            <label class="form-label" for="history-query-description" translate="history-query.description"></label>
+            <textarea formControlName="description" id="history-query-description" class="form-control"></textarea>
+          </div>
+        </div>
+
+        <div class="row mb-3">
+          <label for="start" class="col-auto col-form-label" translate="history-query.start"></label>
+          <div class="col">
+            <oib-datetimepicker formControlName="start" id="start"></oib-datetimepicker>
+          </div>
+          <label for="end" class="col-auto col-form-label" translate="history-query.end"></label>
+          <div class="col">
+            <oib-datetimepicker formControlName="end" id="end"></oib-datetimepicker>
+          </div>
+        </div>
+
+        <!-- Cache settings -->
+        <h2 translate="north.cache-settings.title" id="cache-settings-title"></h2>
+        <div class="row mb-3" formGroupName="caching">
+          <div class="col-4">
+            <oib-scan-mode
+              key="cache-scan-mode"
+              formControlName="scanMode"
+              [scanModes]="scanModes"
+              [label]="'north.cache-settings.send-scan-mode' | translate "
+            ></oib-scan-mode>
+          </div>
+          <!-- Retry interval -->
+          <div class="col-4">
+            <div class="form-group">
```

Comment:
i think we are missing the val-errors and it should be tested as well at least for the number of errors expected